compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java

```diff
@@ -459,12 +457,15 @@ private void generateModuleClasses(BIRPackage module, Map<String, byte[]> jarEnt
             AsyncDataCollector asyncDataCollector = new AsyncDataCollector(moduleClass);
             boolean serviceEPAvailable = false;
             boolean isInitClass = Objects.equals(moduleClass, moduleInitClass);
+            JvmTypeGen jvmTypeGen = new JvmTypeGen(stringConstantsGen);
```

Comment:
Since we need to pass  string constant gen to typegen, I refactor typegen and castgen to have only instance methods and keep string constan tgen gen as class level variable.  With this we do not need to pass string constant gen to all typegen methods. We have done same thing for instructions gen , error gen and terminator gen too.

Reply:
@Kishanthan @manuranga

Reply:
Als o removed asyncCollector pass for evevry method as well.